client/src/components/TaskModal/TagInput.tsx

```diff
@@ -9,44 +9,46 @@ import { Tag } from 'GlobalType';
 axios.defaults.withCredentials = true; // withCredentials 전역 설정
 
 interface TagInputProps {
-  setTagIdx: React.Dispatch<React.SetStateAction<number | null>>;
+  tagObject: Tag | null;
+  setTagObject: React.Dispatch<React.SetStateAction<Tag | null>>;
 }
 
-const TagInput = ({ setTagIdx }: TagInputProps) => {
+const TagInput = ({ tagObject, setTagObject }: TagInputProps) => {
+  const [isTagInputFocused, setIsTagInputFocused] = useState(false);
+  const [tagInput, onChangeTagInput, setTagInput] = useInput('');
   const [tagList, setTagList] = useState<Tag[]>([]);
-  const [tagInput, onTagInputChange, setTagInput] = useInput('');
-  const [selectedTag, setSelectedTag] = useState<Tag | null>(); // 선택된 하나의 태그
-  const [isTagInputFocused, setIsTagInputFocused] = useState<Boolean>(false); // Input창이 Focus될 때만 List 표시
-  const [newTagColor, setNewTagColor] = useState(''); // 새로운 Tag 추가 시 색깔 랜덤 배정
+  const [searchedTagList, setSearchedTagList] = useState<Tag[]>([]);
+  const [newTagColor, setNewTagColor] = useState('');
+  const [reload, setReload] = useState(0);
+
+  //태그 선택 해제
+  const unSetTagItem = () => {
+    setTagObject(null);
+    setTagInput('');
+  };
 
   //TAG GET
   useEffect(() => {
     const fetchData = async () => {
```

Comment:
1. 함수 정의하는 부분을 `useEffect` 안에 넣었을 때 장점이 궁금합니다.
2. 함수 이름을 `httpGetTagList` 정도로 하면 어떨까요
```suggestion
    const httpGetTagList = async () => {
```

Reply:
1. get API를 통해 비동기적으로 데이터를 불러오기 때문에 useEffect로 관리가 필요합니다. 
get으로 가져와야 하는 데이터는 처음 한 번만 불러오면 되기 때문에 주로 axios와 빈 배열의 useEffect로 사용하는게 일반적입니다. [`참고`](https://darrengwon.tistory.com/275)
 해당 코드에서는 데이터를 추가 / 변경 / 삭제 했을 때 다시 불러오는 작업을 위해 reload라는 변수를 추가로 사용했습니다. 
2. 해당 코드에서 post delete 요청에서도 http란 단어를 사용하지 않았기 때문에 일단 getTagList로 변경해 놓았습니다. https://github.com/boostcampwm-2022/web19-Boostart/pull/75/commits/b19a7c5fe784445946b48e937973dedee6b33e5b

Reply:
1. 바깥에서 정의한 다음 호출하는 코드만 `useEffect` 안에 있어도 되지 않나요?
```
const getTagList = () => { ... }

useEffect(() => {
  getTagList();
  ...
```

Reply:
아 넵 그건 상관 없을 거 같습니다.